src/OpenProtocolInterpreter/Tightening Results/Mid0900.cs

```diff
@@ -0,0 +1,303 @@
+
+            Coefficient = coeff;
+            OperationType = op;
+         }
+
+         // defining this at runtime
+         var numberOfResolutionFields = GetField(1, (int)DataFields.NUMBER_OF_RESOLUTION_FIELDS);
+         numberOfResolutionFields.Index = dataFieldListField2.Index + dataFieldListField2.Size;
+         if (!Int32.TryParse(GetValue(numberOfResolutionFields, asciiMessage), out int numberOfResolutionValue))
+            throw new Exception("Cannot parse Number of Resolutions DataField");
+
+
+         // There is a resolution field
+         var resolutionField = GetField(1, (int)DataFields.RESOLUTION_FIELDS);
+         if (numberOfResolutionValue > 0)
+         {
+            // Get field, set index to end of datafieldList2. Set size to end length of the resolutionfields
+            resolutionField.Index = numberOfResolutionFields.Index + numberOfResolutionFields.Size;
+            ResolutionFields = GetResolutionFieldsByFieldCount(asciiMessage.Substring(resolutionField.Index), numberOfResolutionValue, out int endLength);
+            resolutionField.Size = endLength;
+         }
+         else
+         {
+            throw new Exception("No resolution datafield found!");
+         }
+
+         var numberOfTraceSamples = GetField(1, (int)DataFields.NUMBER_OF_TRACE_SAMPLES);
+         numberOfTraceSamples.Index = resolutionField.Index + resolutionField.Size;
+
+         var nullCharacter = GetField(1, (int)DataFields.NUL_CHAR);
+         nullCharacter.Index = numberOfTraceSamples.Index + numberOfTraceSamples.Size;
+
+         // Substring all the fields into their correct property. Once this is called. All the properties are set based on the index and size given
+         ProcessDataFields(asciiMessage);
+         // END OF ASCII DATA. 
+
+         // START OF BINARY DATA
+
+         // Store the raw binary data
+         RawTraceSamples = new byte[package.Length - asciiLength];
+         Array.Copy(package, asciiLength, RawTraceSamples, 0, package.Length - asciiLength);
+
+         // Convert raw binary into 2 byte trace samples
+         TraceSamples = new float[RawTraceSamples.Length / 2];
+         for (int i = 0, j = 0; j < TraceSamples.Length; i += 2, j++)
+         {
+            TraceSamples[j] = ((short)RawTraceSamples[i] << 8 | (ushort)RawTraceSamples[i + 1]);
+         }
+
+         return this;
+      }
+
+
+      public enum DataFields
+      {
+         //rev 1
+         RESULT_DATA_IDENTIFIER = 0,
+         TIME_STAMP = 1,
+         NUMBER_OF_PID_DATA_FIELDS = 2,
+         VARIABLE_DATA_FIELDS = 3,
+         TRACE_TYPE = 4,
+         TRANSDUCER_TYPE = 5,
+         Unit = 6,
+         NUMBER_OF_PID_DATA_FIELDS2 = 7,
+
+         NUMBER_OF_RESOLUTION_FIELDS = 8,
+         RESOLUTION_FIELDS = 9,
+         NUMBER_OF_TRACE_SAMPLES = 10,
+         NUL_CHAR = 11,
+      }
+
+      public static List<VariableDataField> GetDataFieldsByFieldCount(string value, int fieldCount, out int endLength)

```

Comment:
Static methods might be a little tricky when we come up with Multi-threaded applications, is there a possibility to not be static?